### ISSUE TITLE
[no-use-before-define] Error on Nodes 'global' variable

### ISSUE THREAD

**Issue by mf-pherlihy** [typescript-eslint/typescript-eslint#2613]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-use-before-define": [2, { "ignoreTypeReferences": true }]
  }
}
```

```TS
global.someProp = true;

declare global {
  namespace NodeJS {
    interface Global {
      someProp?: boolean;
    }
  }
}
```

**Expected Result**

I would expect no errors since `ignoreTypeReferences` is set to `true`.

**Actual Result**

> 'global' was used before it was defined 

**Additional Info**

I have `@types/node@14.11.2` installed.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.3.0` |
| `@typescript-eslint/parser`        | `4.3.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.10.0` |
| `node`                             | `14.7.0` |

**Comment by yeonjuan** [comment]
@mf-pherlihy 
Hi:) In my case, It's not reproduced. 
Maybe have you define 'global' below? Otherwise, could you share your configuration or code more?

**Comment by mf-pherlihy** [comment]
@yeonjuan -- Yeah you're right, my example was off and I edited my post to reflect the changes.

**Comment by bradzacher** [comment]
The bug here is that the `scope-manager` defines a variable for `declare global`, when it really shouldn't (because it's not a real thing that can be referenced).

It should be a pretty simple fix + a few tests!

Here's where the fix needs to be done:
https://github.com/typescript-eslint/typescript-eslint/blob/cdb9807a5a368a136856cd03048b68e0f2dfb405/packages/scope-manager/src/referencer/Referencer.ts#L767-L773
